qiskit_ibm_provider/transpiler/passes/scheduling/scheduler.py

```diff
@@ -102,14 +106,19 @@ def _visit_node(self, node: DAGNode) -> None:
         # compute t0, t1: instruction interval, note that
         # t0: start time of instruction
         # t1: end time of instruction
-        if isinstance(node.op, self.CONDITIONAL_SUPPORTED):
+        if isinstance(node.op, self.CONDITIONAL_SUPPORTED) and node.op.condition_bits:
             self._visit_conditional_node(node)
         else:
             if node.op.condition_bits:
                 raise TranspilerError(
                     f"Conditional instruction {node.op.name} is not supported in ASAP scheduler."
                 )
 
+            # If True we are coming from a conditional block.
+            # start a new block for the unconditional operations.
+            if self._conditional_block:
```

Comment:
What happens if we input
```python
qc = QuantumCircuit(3, 1)
qc.x(0).c_if(0, True)
qc.x(1)  # <- seems like new block is created here
qc.x(2).c_if(0, True)
```
It seems to me like having all three instructions in a single block should be fine. Perhaps starting new block is not a problem because this program doesn't have barrier.